pkg/controller/master/backup/util.go

```diff
@@ -0,0 +1,117 @@
+package backup
+
+import (
+	"fmt"
+	"time"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	harvesterapiv1 "github.com/rancher/harvester/pkg/apis/harvester.cattle.io/v1alpha1"
+)
+
+const (
+	allNamespaces = ""
+)
+
+func vmBackupReady(vmBackup *harvesterapiv1.VirtualMachineBackup) bool {
+	return vmBackup.Status != nil && vmBackup.Status.ReadyToUse != nil && *vmBackup.Status.ReadyToUse
+}
+
+func vmBackupProgressing(vmBackup *harvesterapiv1.VirtualMachineBackup) bool {
+	return vmBackupError(vmBackup) == nil &&
+		(vmBackup.Status == nil || vmBackup.Status.ReadyToUse == nil || !*vmBackup.Status.ReadyToUse)
+}
+
+func vmBackupError(vmBackup *harvesterapiv1.VirtualMachineBackup) *harvesterapiv1.Error {
+	if vmBackup.Status != nil && vmBackup.Status.Error != nil {
+		return vmBackup.Status.Error
+	}
+	return nil
+}
+
+func vmBackupContentReady(vmBackupContent *harvesterapiv1.VirtualMachineBackupContent) bool {
+	return vmBackupContent.Status != nil && vmBackupContent.Status.ReadyToUse != nil && *vmBackupContent.Status.ReadyToUse
+}
+
+func getVMBackupContentName(vmBackup *harvesterapiv1.VirtualMachineBackup) string {
+	if vmBackup.Status != nil && vmBackup.Status.VirtualMachineBackupContentName != nil {
+		return *vmBackup.Status.VirtualMachineBackupContentName
+	}
+
+	return fmt.Sprintf("%s-%s", vmBackup.Name, "backupcontent")
+}
+
+func newError(message string) *harvesterapiv1.Error {
+	return &harvesterapiv1.Error{
+		Message: &message,
+		Time:    currentTime(),
+	}
+}
+
+func newReadyCondition(status corev1.ConditionStatus, reason string) harvesterapiv1.Condition {
+	return harvesterapiv1.Condition{
+		Type:               harvesterapiv1.BackupConditionReady,
+		Status:             status,
+		Reason:             reason,
```

Comment:
Probably use message instead of reason
https://github.com/kubernetes/kubernetes/blob/bce5d7a0158444080ed95aae6a5684ce4dd9444b/pkg/apis/core/types.go#L3166-L3171